src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2057,6 +2102,19 @@ private bool HasArrayConversionToInterface(ArrayTypeSymbol source, TypeSymbol de
             return HasIdentityOrImplicitReferenceConversion(source.ElementType.TypeSymbol, argument0, ref useSiteDiagnostics);
         }
 
+        private bool HasImplicitReferenceConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            if (IncludeNullability)
+            {
+                if (!HasTopLevelNullabilityImplicitConversion(source, destination))
+                {
+                    return false;
+                }
+                return HasIdentityOrImplicitReferenceConversion(source.TypeSymbol, destination.TypeSymbol, ref useSiteDiagnostics);
```

Comment:
>return HasIdentityOrImplicitReferenceConversion(source.TypeSymbol, destination.TypeSymbol, ref useSiteDiagnostics); [](start = 16, length = 115)

Why not simply fall through?

Reply:
This case allows for identity when ignoring nullability. For instance, when `source` is `string` and `destination` is `string?`, `HasImplicitReferenceConversion` should return `true` when nullability is considered and `false` otherwise.

---
In reply to: [168038301](https://github.com/dotnet/roslyn/pull/23981#discussion_r168038301) [](ancestors = 168038301)